server/etcdserver/server_test.go

```diff
@@ -198,6 +198,115 @@ func TestV2SetClusterVersion(t *testing.T) {
 	}
 }
 
+func TestApplyConfStateWithRestart(t *testing.T) {
+	n := newNodeRecorder()
+	srv := newServer(t, n)
+
+	assert.Equal(t, srv.consistIndex.ConsistentIndex(), uint64(0))
+
+	var nodeID uint64 = 1
+	memberAddContext, err := json.Marshal(&membership.Member{ID: types.ID(nodeID), RaftAttributes: membership.RaftAttributes{PeerURLs: []string{""}}})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	entries := []raftpb.Entry{
+		{
+			Term:  1,
+			Index: 1,
+			Type:  raftpb.EntryConfChange,
+			Data: pbutil.MustMarshal(&raftpb.ConfChange{
+				Type:    raftpb.ConfChangeAddNode,
+				NodeID:  nodeID,
+				Context: memberAddContext,
+			}),
+		},
+		{
+			Term:  1,
+			Index: 2,
+			Type:  raftpb.EntryConfChange,
+			Data: pbutil.MustMarshal(&raftpb.ConfChange{
+				Type:   raftpb.ConfChangeRemoveNode,
+				NodeID: nodeID,
+			}),
+		},
+		{
+			Term:  1,
+			Index: 3,
+			Type:  raftpb.EntryConfChange,
+			Data: pbutil.MustMarshal(&raftpb.ConfChange{
+				Type:    raftpb.ConfChangeAddNode,
+				NodeID:  nodeID,
```

Comment:
Note that etcd generates a new memberId each time when users add a new member. It isn't valid to use the same nodeID.

Reply:
True, updated to use ConfChangeUpdateNode